.github/workflows/commit-checker.yml

```diff
@@ -0,0 +1,42 @@
+name: Commit checker
+
+on:
+  pull_request:
+
+jobs:
+  commit-checker:
+    name: Commit checker
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: 3
+
+    - name: Get pull-request commits
+      run: |
+        set -x
+        # We are in a merge, HEAD^ is the base branch, HEAD^2 is the merged branch
+        # Get the latest commit from the pull-request in history. If no parent is found for the
+        # latest commit it may not be the latest, increase history
+        while [ ! $(git rev-parse --verify `git rev-list --reverse HEAD^..HEAD^2 | head -n1`^) ]; do
```

Comment:
`while [ -z "$(git merge-base HEAD^ HEAD^2)" ]; do`
in general, please don't use `!` for non-numbers, and don't use the ` shell escape :D